frame/support/src/traits.rs

```diff
@@ -211,6 +209,13 @@ pub trait Contains<T: Ord> {
 	fn add(t: &T) { unimplemented!() }
 }
 
+/// A trait for querying an upper bound on the number of element in a type that implements
+/// `Contains`.
+pub trait ContainsCountUpperBound {
```

Comment:
```suggestion
pub trait ContainsLengthBound {
    fn min_len() -> usize;
    fn max_len() -> usize;
}    
```

not sure if it works but sounds more ergonomic to me.

Reply:
yep done